include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,918 @@
+    {
+        if (begin_it >= end_it) // [[unlikely]]
+            return begin() + std::ranges::distance(cbegin(), end_it);
+
+        size_type const length = std::ranges::distance(begin_it, end_it);
+        auto out_it = begin() + std::ranges::distance(cbegin(), begin_it);
+
+        while (end_it != cend())
+            *(out_it++) = *(end_it++);
+
+        sz -= length;
+        return begin() + std::ranges::distance(cbegin(), begin_it);
+    }
+
+    /*!\brief Removes specified elements from the container.
+     * \param   pos Remove the element at pos.
+     * \returns     Iterator following the last element removed. If the iterator `pos` refers to the last element,
+     *              the end() iterator is returned.
+     *
+     * Invalidates iterators and references at or after the point of the erase, including the end() iterator.
+     *
+     * The iterator `pos` must be valid and dereferenceable. Thus the end() iterator (which is valid, but is not
+     * dereferencable) cannot be used as a value for pos.
+     *
+     * ### Complexity
+     *
+     * Linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container may contain invalid data after exception is
+     * thrown.
+     */
+    constexpr iterator erase(const_iterator pos)
+    {
+       return erase(pos, pos + 1);
+    }
+
+    /*!\brief Appends the given element value to the end of the container.
+     * \param value The value to append.
+     *
+     * If the new size() is greater than capacity() this is undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container may contain invalid data after exception is
+     * thrown.
+     */
+    constexpr void push_back(value_type const value)
+    {
+        data[sz] = value;
+        ++sz;
+    }
+
+    /*!\brief Removes the last element of the container.
+     *
+     * Calling pop_back() on an empty container is undefined. In debug mode an assertion will be thrown.
+     *
+     * No iterators or references except for back() and end() are invalidated.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No exception is thrown in release mode.
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    constexpr void pop_back()
+    {
+        --sz;
+    }
+
+    /*!\brief Resizes the container to contain count elements.
+     * \param[in] count The new size.
+     *
+     * If count is greater than capacity this is undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr void resize(size_type const count)
+    {
+        sz = count;
+    }
```

Comment:
Why throw an exception? Doesn't an assert suffice?
std::vector does not throw if new size exceeds max size.

Reply:
> std::vector does not throw if new size exceeds max size.

Yes, but max_size on a vector is also never reached in practice. I suspect it will happen a lot more often for this data structure, because it's the central "limitation".  Since resize is not called that often and regular container semantics imply that it's not fast anyway, I think it wouldn't hurt to throw an exception here.

Reply:
except that this function cannot be constexpr anymore then

Reply:
that's right :cry: 
Ok, let's only assert for now.